.github/workflows/update_goldens.yml

```diff
@@ -0,0 +1,41 @@
+name: Update Goldens
+on: 
+  workflow_dispatch:
+    inputs:
+      commit:
+        type: boolean
+        default: true
+        required: true
+        description: Commit new Goldens to this branch.
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: 📚 Git Checkout
+        uses: actions/checkout@v3
+
+      - name: 🏗 Build and test Sweyer
+        uses: ./.github/workflows/build_and_test
+        with:
+          testing_arguments: --update-goldens
+
+      - name: 📁 Save updated Golden test artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: golden-test-updated
+          path: test/golden/goldens
+
+      - name: 💂 Verify that ${{ github.actor }} has permission to commit
+        if: github.event.inputs.commit == 'true'
+        id: permissionCheck
+        uses: actions-cool/check-user-permission@v2
+        with:
+          require: write
+      - name: 💾 Commit updated Golden test artifacts
+        uses: stefanzweifel/git-auto-commit-action@v4
+        if: github.event.inputs.commit == 'true' && steps.permissionCheck.outputs.require-result == 'true'
```

Comment:
The workflow should fail if it doesn't have permissions to commit

So it's better to remove this entire thing witrh `permissionCheck`

Reply:
Ok, I think it is actually perfect that you can only run the workflow as a collaborator, because you can always run it on your own local fork. No need to run it here.

I will update the contributing documentation accordingly. Please merge this into `1.0.8` as well afterwards.